sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -743,6 +743,37 @@ def test_precompute_invalid_argument():
                         "Got 'auto'", ElasticNet(precompute='auto').fit, X, y)
 
 
+def test_precompute_incorrect_gram():
+    X, y, _, _ = build_dataset()
+
+    X_centered = X - np.average(X, axis=0)
+    garbage = np.random.normal(size=X.shape)
+    precompute = np.dot(garbage.T, garbage)
+
+    clf = ElasticNet(alpha=0.01, precompute=precompute)
+
+    assert_raises_regex(ValueError, "Gram matrix.*did not pass validation.*",
+                        clf.fit, X_centered, y)
+
+
+def test_precompute_gram_weighted_samples():
+    X, y, _, _ = build_dataset()
+    sample_weight = np.random.lognormal(size=y.shape)
```

Comment:
please use a fixed RandomState

Reply:
still to do.

also we tend to call instances of RandomState `rng`

Reply:
ah, missed the second one. Should be fixed now.